### PR TITLE
DB-9624 Ensure sqlshell.sh works on Windows

### PR DESCRIPTION
**Pull request by hatyo**
Ready for CR.

### CODE REVIEW

File: assembly/pom.xml [hunk 1]
```diff
@@ -240,6 +240,9 @@
                                         <copy todir="target/SPLICEMACHINE-${project.version}.${envClassifier}.${parcel.patch}/bin">
                                             <fileset dir="common/src/main/resources/bin" includes="sqlshell.sh"/>
                                         </copy>
+                                        <copy todir="target/SPLICEMACHINE-${project.version}.${envClassifier}.${parcel.patch}/bin">
```
Comment (OlegMazurov) [inline assembly/pom.xml:243]: I don't think we should include _winsqlshell.cmd_ into Cloudera parcel.

File: assembly/pom.xml [hunk 2]
```diff
@@ -644,6 +647,9 @@
                                                 <source>
                                                     <location>common/src/main/resources/bin/sqlshell.sh</location>
                                                 </source>
+                                                <source>
```
Comment (OlegMazurov) [inline assembly/pom.xml:650]: ditto

File: assembly/pom.xml [hunk 3]
```diff
@@ -1188,6 +1194,9 @@
...
```
Comment (OlegMazurov) [inline assembly/pom.xml:1197]: ditto

File: assembly/pom.xml (hunk 1 above)
Comment (hatyo) [reply assembly/pom.xml:243]: You're right, I thought we would ship it with all supported deliverables, but this is incorrect, we will just add it to the `sqlshell` profile. Thank you for the hint.

File: assembly/pom.xml (hunk 2 above)
Comment (hatyo) [reply assembly/pom.xml:650]: ok :)

File: GETTING-STARTED.md [hunk 4]
```diff
@@ -66,3 +66,7 @@ This will compile everything and start the database.
 Then to connect to the database:
 
 `./sqlshell.sh`
+
+Or if you are on Windows:
+
+`call sqlshell.cmd`
```
Comment (OlegMazurov) [inline GETTING-STARTED.md:72]: See comment regarding sqshell.sh below.

File: sqlshell.cmd [hunk 5]
```diff
@@ -0,0 +1,14 @@
+@echo off
```
Comment (OlegMazurov) [inline sqlshell.cmd:1]: That would be a Splice developer sqlshell on Windows. Only there are no Splice developers on Windows.
I think this file should be removed for now.

Comment (hatyo) [reply sqlshell.cmd:1]: You're right, I will remove it (and revert the comment in GETTING-STARTED.md

File: GETTING-STARTED.md (hunk 4 above)
Comment (hatyo) [reply GETTING-STARTED.md:72]: ok

**Comment by hatyo** [comment]
OK Oleg, thanks for the comments, I am pushing a new commit addressing them

**Comment by hatyo** [comment]
I made a stupid mistake in the branch name (DB9624 instead of DB-9624). If necessary, I can definitely create a new PR using the correct naming format.

**Comment by hatyo** [comment]
Thanks a lot @OlegMazurov , can I bring in this change or should I ask Jenkins to verify it first? (although this is probably not necessary since no business logic code is touched)

File: assembly/common/src/main/resources/bin/winsqlshell.cmd [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+   REM error, print help and exit
+   CALL :help
+   EXIT /B 1
+)
+
+REM set splice lib dir, for Windows we assume it is always local
+
+set SCRIPT_DIR=%SCRIPT_DIR:"=%
+SET SPLICE_LIB_DIR=%SCRIPT_DIR%lib
+echo SPLICE_LIB_DIR=%SPLICE_LIB_DIR%
```
Comment (martinrupp) [inline assembly/common/src/main/resources/bin/winsqlshell.cmd:145]: needed?

File: assembly/common/src/main/resources/bin/winsqlshell.cmd [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+    -Djavax.net.ssl.trustStore.ssl.trustStorePassword^=%CLIENT_SSL_TRUSTSTOREPASSWD%
+)
+
+call :message Running Splice Machine SQL shell
+call :message For help%: "splice> help;"
+REM todo: catch stderr messages and filter out Picked up JAVA_TOOL_OPTIONS   2>nul
+java -classpath %CLASSPATH% %GEN_SYS_ARGS% %SSL_ARGS% %IJ_SYS_ARGS% com.splicemachine.db.tools.ij %SCRIPT% 2>nul
+
+CALL :eof
+
```
Comment (martinrupp) [inline assembly/common/src/main/resources/bin/winsqlshell.cmd:245]: this is returning and then printing ECHO %* again, resulting e.g. in
```
splice> exit;
-h srv131 -u bla -s password
```
at the end.

File: assembly/common/src/main/resources/bin/winsqlshell.cmd (hunk 6 above)
Comment (hatyo) [reply assembly/common/src/main/resources/bin/winsqlshell.cmd:145]: First two lines are needed, last echo can be removed. I removed it.

File: assembly/common/src/main/resources/bin/winsqlshell.cmd (hunk 7 above)
Comment (hatyo) [reply assembly/common/src/main/resources/bin/winsqlshell.cmd:245]: Right, thanks, fixed it.

**Comment by hatyo** [comment]
> 
> 
> additionally it would be cool to have a super basic tutorial how to use (can be also a wiki page).
> like
> 
>     * make sure you have java installed (Java JRE is enough), e.g. go Command Prompt, enter `java`
> 
>     * extract file
> 
>     * go to command shell, cd sqlshell
>       `winsqlshell.cmd -h yourhost -u splice -P`

There is already a How-To page explaining how to use this interactive SQL shell in Windows. Moreover, similar to `sqlshell.sh`, there is a check for not only the existence of Java in PATH but also that its version >=1.8.

**Comment by hatyo** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

**Pull request merged**